grudge/trace_pair.py

```diff
@@ -432,6 +434,11 @@ def finish(self):
                          exterior=bdry_conn(self.remote_data))
 
 
+class _TagKeyBuilder(KeyBuilder):
+    def update_for_type(self, key_hash, key: Type[Any]):
+        self.rec(key_hash, (key.__module__, key.__name__, key.__name__,))
```

Comment:
Out of curiosity, do you remember why `key.__name__` was included twice here @inducer?

Reply:
No, I don't see a good reason for that.